dev/clint/src/clint/linter.py

```diff
@@ -263,11 +277,34 @@ def _syntax_error_example(self, node: ast.FunctionDef | ast.AsyncFunctionDef) ->
                         docstring_node.lineno + code_block.loc.lineno,
                         code_block.loc.col_offset,
                     )
-                    self._check(loc, EXAMPLE_SYNTAX_ERROR)
+                    self._check(loc, rules.ExampleSyntaxError())
+
+    def _param_mismatch(self, node: ast.FunctionDef | ast.AsyncFunctionDef) -> None:
+        if node.name.startswith("_"):
+            return
+        if docstring_node := self._docstring(node):
+            if (doc_args := _parse_docstring_args(docstring_node.value)) and (
+                func_args := self._parse_func_args(node)
+            ):
+                func_args_set = set(func_args)
+                doc_args_set = set(doc_args)
+                # TODO: Enable this rule
+                # if diff := func_args_set - doc_args_set:
+                #     self._check(Location.from_node(node), rules.MissingDocstringParam(list(diff)))
+
+                if diff := doc_args_set - func_args_set:
+                    self._check(
+                        Location.from_node(node), rules.ExtraneousDocstringParam(list(diff))
+                    )
+
+                # TODO: Enable this rule
+                # if func_args_set == doc_args_set and func_args != doc_args:
+                #     self._check(Location.from_node(node), rules.DocstringParamOrder())
```

Comment:
@Abubakar17 Let's fix this issue next. I'll merge this PR once all the CI checks successfully finish.

Reply:
you can uncomment this block and file a PR, then the Lint check should tell you where to fix.